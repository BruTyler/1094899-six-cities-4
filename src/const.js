export const PlaceType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`,
};
