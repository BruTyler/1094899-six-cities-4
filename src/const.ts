export enum PlaceType {
  APARTMENT = `Apartment`,
  ROOM = `Private room`,
  HOUSE = `House`,
  HOTEL = `Hotel`,
};

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW = `Price: low to high`,
  PRICE_HIGH = `Price: high to low`,
  TOP_RATED = `Top rated first`,
};

export enum AuthorizationStatus {
  NO_AUTH = `NO_AUTH`,
  AUTH = `AUTH`,
};

export enum BuisnessRequirements {
  MAX_PHOTOS_PER_APARTMENT = 6,
  MAX_REVIEWS_PER_APARTMENT = 10,
  MIN_REVIEW_TEXT_LENGTH = 50,
  MAX_REVIEW_TEXT_LENGTH = 300,
};

export enum AppRoute {
  AUTH = `/login`,
  ROOT = `/`,
  FAVORITES = `/favorites`,
  PROPERTY_WITH_ID = `/offer/:id`,
};

export enum ApartmentEnvironment {
  FAVORITE = `favorites`,
  NEARBY_PLACES = `near-places`,
  MAIN_WINDOW = `cities`
};

export enum MapEnvironment {
  NEARBY_PLACES = `property__map`,
  MAIN_WINDOW = `cities__map`
};
