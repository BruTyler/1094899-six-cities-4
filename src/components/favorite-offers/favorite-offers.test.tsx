import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import FavoriteOffers from './favorite-offers';
import {AuthorizationStatus} from '../../const';
import NameSpace from '../../reducer/name-space';

const CITIES = [
  {
    id: `A`,
    location: [1, 1],
    defaultZoom: 1
  },
  {
    id: `B`,
    location: [2, 2],
    defaultZoom: 2
  },
];

const APARTMENTS = [
  {
    id: 0,
    cityId: `A`,
    type: `Apartment`,
    description: `description0`,
    rating: 0.1,
    price: 1,
    isPremium: true,
    isFavourite: true,
    photo: `img0.jpg`,
    location: [1, 2],
  },
  {
    id: 1,
    cityId: `A`,
    type: `Private room`,
    description: `description1`,
    rating: 1,
    price: 2,
    isPremium: false,
    isFavourite: false,
    photo: `img1.jpg`,
    location: [3, 4],
  },
  {
    id: 2,
    cityId: `B`,
    type: `Private room`,
    description: `description3`,
    rating: 3,
    price: 3,
    isPremium: false,
    isFavourite: false,
    photo: `img2.jpg`,
    location: [5, 6],
  },
];

const mockStore = configureStore();

describe(`<FavoriteOffers /> render suit`, () => {
  it(`<FavoriteOffers /> render favorite apartment list case`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });
    const generatedTree = renderer.create(
        <Provider store={store}>
          <FavoriteOffers
            favoriteCities={CITIES}
            favoriteApartments={APARTMENTS}
            onCityTitleClick={() => {}}
          />
        </Provider>
    ).toJSON();

    expect(generatedTree).toMatchSnapshot();
  });
});
