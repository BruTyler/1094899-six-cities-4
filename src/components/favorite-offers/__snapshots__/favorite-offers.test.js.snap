// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteOffers /> render suit <FavoriteOffers /> render favorite apartment list case 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites"
  >
    <h1
      className="favorites__title"
    >
      Saved listing
    </h1>
    <ul
      className="favorites__list"
    >
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="#"
              onClick={[Function]}
            >
              <span>
                A
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="favorites__place-card favorites__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="description0"
                  className="place-card__image"
                  height="110"
                  src="img0.jpg"
                  width="150"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    1
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button "
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In
                     bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  description0
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
          <article
            className="favorites__place-card favorites__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="description1"
                  className="place-card__image"
                  height="110"
                  src="img1.jpg"
                  width="150"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    2
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button  place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To
                     bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "20%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  description1
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Private room
              </p>
            </div>
          </article>
        </div>
      </li>
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="#"
              onClick={[Function]}
            >
              <span>
                B
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="favorites__place-card favorites__card place-card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="favorites__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="description3"
                  className="place-card__image"
                  height="110"
                  src="img2.jpg"
                  width="150"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    3
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button  place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To
                     bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  description3
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Private room
              </p>
            </div>
          </article>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
